tests/test_cli.py

```diff
@@ -201,3 +201,99 @@ def test_aggregate_forcings_to_hrus(self, tmp_path):
         assert abs(val[16071, 0] - 0.569977) < 1e-04
         assert abs(val[0, 50] - 0.010276) < 1e-04
         assert abs(val[16071, 50] - 0.516639) < 1e-04
+
+    def test_aggregate_forcings_to_hrus_with_nodata(self, tmp_path):
+        runner = CliRunner()
+        output_nc_file_path = tmp_path / "aggreg.nc"
+        output_weight_file_path = tmp_path / "weight_aggreg.rvt"
+        params = [
+            get_local_testdata("raven-routing-sample/VIC_test_nodata.nc"),
+            get_local_testdata("raven-routing-sample/VIC_test_nodata_weights.rvt"),
+            "-v",
+            "et",
+            "--dim-names",
+            "rlon",
+            "rlat",
+            "--output-nc-file",
+            output_nc_file_path,
+            "--output-weight-file",
+            output_weight_file_path,
+        ]
+        params = map(str, params)
+
+        result = runner.invoke(aggregate_forcings_to_hrus, params)
+
+        assert result.exit_code == 0
+        assert not result.exception
+
+        output_rvt = output_weight_file_path.read_text()
+
+        gws = GridWeightsCommand.parse(output_rvt)
+
+        new_weights = gws.data
+
+        # check new weights
+        assert new_weights[0][0] == 1  # These are the HRU-IDs
+        assert new_weights[1][0] == 2  # These are the HRU-IDs
+        assert new_weights[2][0] == 3  # These are the HRU-IDs
+
+        assert new_weights[0][1] == 0  # These need to be exactly [0,1,2,3,...,nHRU]
+        assert new_weights[1][1] == 1  # These need to be exactly [0,1,2,3,...,nHRU]
+        assert new_weights[2][1] == 2  # These need to be exactly [0,1,2,3,...,nHRU]
+
+        assert new_weights[0][2] == 1.0  # All new_weights[:][2] need to be 1.0
+        assert new_weights[1][2] == 1.0  # All new_weights[:][2] need to be 1.0
+        assert new_weights[2][2] == 1.0  # All new_weights[:][2] need to be 1.0
+
```

Comment:
Would `numpy.testing.assert_array_equal` be useful here ? e.g. something like
`np.testing.assert_array_equal(new_weights[:n, 0], range(1,n+1))`